roles/job/templates/job_definition.yml.j2

```diff
@@ -18,8 +18,14 @@ spec:
               value: "{{ tower_config_secret['resources'][0]['data']['token'] | b64decode }}"
             - name: TOWER_HOST
               value: "{{ tower_config_secret['resources'][0]['data']['host'] | b64decode }}"
+{% if job_template_name is defined %}
             - name: TOWER_JOB_TEMPLATE_NAME
               value: "{{ job_template_name }}"
+{% endif %}
+{% if workflow_template_name is defined %}
+            - name: TOWER_WORKFLOW_TEMPLATE_NAME
+              value: "{{ workflow_template_name }}"
+{% endif %}
```

Comment:
We should add an assertion that at least one of the two variables is defined before trying to apply the job_definition.yml.  

```
- name: Assert JT or WFJT defined
  assert:
    that:
      - (job_template_name | default('') | length) or (workflow_template_name | default('') | length)
  ```

Reply:
Probably right before this task - https://github.com/ansible/awx-resource-operator/blob/devel/roles/job/tasks/main.yml#L102

Reply:
Wouldn't we just want that at the beginning of the main.yml since the whole run is going to fail anyways if neither of those are defined?

Reply:
good point, yes.

Reply:
Done: https://github.com/ansible/awx-resource-operator/pull/70/files#diff-3cf64a57f211f6c16c0451aec752ec3fceb7f144db451d3cfc90a4fc1de5010aR3